INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 09:38:21] "[33mGET / HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 09:38:21] "[33mGET /favicon.ico HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 09:38:31] "[33mGET / HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 09:38:31] "[33mGET /favicon.ico HTTP/1.1[0m" 404 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:32:45] "[33mGET / HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:32:46] "[33mGET / HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:32:47] "[33mGET / HTTP/1.1[0m" 404 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
ERROR:app:Exception on / [GET]
Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\producto\productos.py", line 12, in Inicio
    return render_template('productoIndex.html', productos = productos)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\templating.py", line 152, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\templating.py", line 133, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\producto\templates\productoIndex.html", line 11, in top-level template code
    <a href="{{url_for('appInicio.Inicio')}}"><button class="boton">Inicio</button></a>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1071, in url_for
    return self.handle_url_build_error(error, endpoint, values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1060, in url_for
    rv = url_adapter.build(  # type: ignore[union-attr]
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\routing\map.py", line 919, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'appInicio.Inicio'. Did you mean 'appProducto.Inicio' instead?
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:33:26] "[35m[1mGET / HTTP/1.1[0m" 500 -
ERROR:app:Exception on / [GET]
Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\producto\productos.py", line 12, in Inicio
    return render_template('productoIndex.html', productos = productos)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\templating.py", line 152, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\templating.py", line 133, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\producto\templates\productoIndex.html", line 11, in top-level template code
    <a href="{{url_for('appInicio.Inicio')}}"><button class="boton">Inicio</button></a>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1071, in url_for
    return self.handle_url_build_error(error, endpoint, values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1060, in url_for
    rv = url_adapter.build(  # type: ignore[union-attr]
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\routing\map.py", line 919, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'appInicio.Inicio'. Did you mean 'appProducto.Inicio' instead?
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:33:29] "[35m[1mGET / HTTP/1.1[0m" 500 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:35:20] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:35:20] "[33mGET /static/index.css HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:35:24] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:35:24] "[33mGET /static/index.css HTTP/1.1[0m" 404 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:37:50] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:37:50] "[33mGET /static/index.css HTTP/1.1[0m" 404 -
ERROR:app:Exception on /producto/agregar [GET]
Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\producto\productos.py", line 24, in Agregar
    return render_template('productoAgregar.html', newproducto = newproducto)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\templating.py", line 152, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\templating.py", line 133, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\producto\templates\productoAgregar.html", line 13, in top-level template code
    {{newProducto.csrf_token}}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'newProducto' is undefined
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:37:58] "[35m[1mGET /producto/agregar HTTP/1.1[0m" 500 -
ERROR:app:Exception on /producto/agregar [GET]
Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\producto\productos.py", line 24, in Agregar
    return render_template('productoAgregar.html', newproducto = newproducto)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\templating.py", line 152, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\templating.py", line 133, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\producto\templates\productoAgregar.html", line 13, in top-level template code
    {{newProducto.csrf_token}}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'newProducto' is undefined
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:38:02] "[35m[1mGET /producto/agregar HTTP/1.1[0m" 500 -
ERROR:app:Exception on /producto/agregar [GET]
Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\producto\productos.py", line 24, in Agregar
    return render_template('productoAgregar.html', newproducto = newproducto)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\templating.py", line 152, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\templating.py", line 133, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\producto\templates\productoAgregar.html", line 13, in top-level template code
    {{newProducto.csrf_token}}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\jinja2\environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'newProducto' is undefined
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:38:04] "[35m[1mGET /producto/agregar HTTP/1.1[0m" 500 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:39:27] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:39:27] "[33mGET /static/index.css HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:39:27] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:39:27] "[33mGET /static/index.css HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:39:28] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:39:28] "[33mGET /static/index.css HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:39:29] "GET /producto/agregar HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:39:29] "[33mGET /static/formulario.css HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:40:51] "[32mPOST /producto/agregar HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:40:51] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:40:51] "[33mGET /static/index.css HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:40:55] "GET /producto/editar/1 HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:40:55] "[33mGET /static/formulario.css HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:40:56] "[32mPOST /producto/editar/1 HTTP/1.1[0m" 302 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:40:56] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:40:56] "[33mGET /static/index.css HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:47:52] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 10:47:52] "[33mGET /static/index.css HTTP/1.1[0m" 404 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:04:00] "[33mGET /inventario/json/index HTTP/1.1[0m" 404 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:04:55] "GET /inventario/json/index HTTP/1.1" 200 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:07:29] "GET /inventario/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:10:21] "GET /inventario/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:10:22] "GET /inventario/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:10:22] "GET /inventario/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:10:23] "GET /inventario/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:10:23] "GET /inventario/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:10:23] "GET /inventario/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:10:38] "[31m[1mPOST /inventario/json/index HTTP/1.1[0m" 405 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:10:42] "GET /inventario/json/index HTTP/1.1" 200 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
ERROR:app:Exception on /inventario/json/index [GET]
Traceback (most recent call last):
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 10, in Inicio
    json = request.get_json()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 21, in Inicio
    return jsonify({'status':400, 'mensaje':ex})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\__init__.py", line 170, in jsonify
    return current_app.json.response(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 215, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 180, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 120, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type UnsupportedMediaType is not JSON serializable
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:19:49] "[35m[1mGET /inventario/json/index HTTP/1.1[0m" 500 -
ERROR:app:Exception on /inventario/json/index [GET]
Traceback (most recent call last):
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 10, in Inicio
    json = request.get_json()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 21, in Inicio
    return jsonify({'inventarios' : listDepa})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\__init__.py", line 170, in jsonify
    return current_app.json.response(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 215, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 180, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 120, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type UnsupportedMediaType is not JSON serializable
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:22:24] "[35m[1mGET /inventario/json/index HTTP/1.1[0m" 500 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
ERROR:app:Exception on /inventario/json/index [GET]
Traceback (most recent call last):
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 10, in Inicio
    json = request.get_json()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 23, in Inicio
    return jsonify({'status':400, 'mensaje':ex})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\__init__.py", line 170, in jsonify
    return current_app.json.response(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 215, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 180, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 120, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type UnsupportedMediaType is not JSON serializable
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:22:29] "[35m[1mGET /inventario/json/index HTTP/1.1[0m" 500 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on http://127.0.0.1:5000
INFO:werkzeug:[33mPress CTRL+C to quit[0m
ERROR:app:Exception on /inventario/json/index [GET]
Traceback (most recent call last):
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 10, in Inicio
    json = request.get_json()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 23, in Inicio
    return jsonify({'status':400, 'mensaje':ex})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\__init__.py", line 170, in jsonify
    return current_app.json.response(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 215, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 180, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 120, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type UnsupportedMediaType is not JSON serializable
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:22:57] "[35m[1mGET /inventario/json/index HTTP/1.1[0m" 500 -
ERROR:app:Exception on /inventario/json/index [GET]
Traceback (most recent call last):
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 10, in Inicio
    json = request.get_json()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 23, in Inicio
    return jsonify({'status':400, 'mensaje':ex})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\__init__.py", line 170, in jsonify
    return current_app.json.response(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 215, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 180, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 120, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type UnsupportedMediaType is not JSON serializable
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:22:58] "[35m[1mGET /inventario/json/index HTTP/1.1[0m" 500 -
ERROR:app:Exception on /inventario/json/index [GET]
Traceback (most recent call last):
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 10, in Inicio
    json = request.get_json()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 23, in Inicio
    return jsonify({'status':400, 'mensaje':ex})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\__init__.py", line 170, in jsonify
    return current_app.json.response(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 215, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 180, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 120, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type UnsupportedMediaType is not JSON serializable
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:22:58] "[35m[1mGET /inventario/json/index HTTP/1.1[0m" 500 -
ERROR:app:Exception on /inventario/json/index [GET]
Traceback (most recent call last):
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 10, in Inicio
    json = request.get_json()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 23, in Inicio
    return jsonify({'status':400, 'mensaje':ex})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\__init__.py", line 170, in jsonify
    return current_app.json.response(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 215, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 180, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 120, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type UnsupportedMediaType is not JSON serializable
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:22:58] "[35m[1mGET /inventario/json/index HTTP/1.1[0m" 500 -
ERROR:app:Exception on /inventario/json/index [GET]
Traceback (most recent call last):
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 10, in Inicio
    json = request.get_json()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 23, in Inicio
    return jsonify({'status':400, 'mensaje':ex})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\__init__.py", line 170, in jsonify
    return current_app.json.response(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 215, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 180, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 120, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type UnsupportedMediaType is not JSON serializable
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:22:58] "[35m[1mGET /inventario/json/index HTTP/1.1[0m" 500 -
ERROR:app:Exception on /inventario/json/index [GET]
Traceback (most recent call last):
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 10, in Inicio
    json = request.get_json()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 23, in Inicio
    return jsonify({'status':400, 'mensaje':ex})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\__init__.py", line 170, in jsonify
    return current_app.json.response(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 215, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 180, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 120, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type UnsupportedMediaType is not JSON serializable
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:22:58] "[35m[1mGET /inventario/json/index HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:23:54] "GET /proveedor/json/index HTTP/1.1" 200 -
ERROR:app:Exception on /inventario/json/index [GET]
Traceback (most recent call last):
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 10, in Inicio
    inventarios = Inventario.query.all()
           ^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\wrappers.py", line 130, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\routes\inventario\inventarios.py", line 23, in Inicio
    return jsonify({'status': 400, 'mensaje': str(ex)})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\__init__.py", line 170, in jsonify
    return current_app.json.response(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 215, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 180, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jonat\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "c:\Users\jonat\Desktop\MULTIPARADIGMA\MultiparadigmaRepo\Unidad 2 - Examen\exa_flask\.venv\Lib\site-packages\flask\json\provider.py", line 120, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type UnsupportedMediaType is not JSON serializable
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:24:40] "[35m[1mGET /inventario/json/index HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:26:59] "[33mGET /proveedor/json/agregar HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:27:13] "[33mPOST /proveedor/json/agregar HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:27:52] "[31m[1mPOST /sucursal/json/index HTTP/1.1[0m" 405 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:27:54] "[31m[1mPOST /sucursal/json/index HTTP/1.1[0m" 405 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:27:56] "GET /sucursal/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:28:06] "GET /venta/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:29:01] "GET /sucursal/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:30:05] "GET /sucursal/json/index HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:30:08] "[33mGET /ventas/json/index HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Nov/2023 11:30:17] "GET /venta/json/index HTTP/1.1" 200 -
